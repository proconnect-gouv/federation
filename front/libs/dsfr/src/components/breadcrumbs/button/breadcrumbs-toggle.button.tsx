import { useToggle } from 'usehooks-ts';

import { t } from '@fc/i18n';

interface BreadCrumbsToggleButtonProps {
  id: string;
}

export const BreadCrumbsToggleButton = ({ id }: BreadCrumbsToggleButtonProps) => {
  const [show, toggleShow] = useToggle(false);

  return (
    <button
      aria-controls={id}
      aria-expanded={show}
      className="fr-breadcrumb__button"
      onClick={toggleShow}>
      {t('DSFR.breadcrumbs.show')}
    </button>
  );
};

BreadCrumbsToggleButton.displayName = 'BreadCrumbsToggleButton';
