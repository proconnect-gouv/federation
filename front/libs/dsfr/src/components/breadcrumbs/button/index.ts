export * from './breadcrumbs-toggle.button';
