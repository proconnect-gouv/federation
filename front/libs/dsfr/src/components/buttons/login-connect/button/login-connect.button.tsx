import classnames from 'classnames';
import React from 'react';

import type { PropsWithOnClick } from '@fc/common';
import { t } from '@fc/i18n';

import type { ButtonTypes } from '../../../../enums';
import { ConnectTypes } from '../../../../enums';
import styles from './login-connect.button.module.scss';

interface LoginConnectButtonProps extends PropsWithOnClick {
  type: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({ connectType, onClick, type }: LoginConnectButtonProps) => {
    const loginLabel = t('DSFR.button.loginConnect.authenticateWith');
    return (
      <button
        className={classnames('fr-connect', {
          [styles['fr-connect--proconnect']]: connectType === ConnectTypes.PRO_CONNECT,
        })}
        type={type}
        onClick={onClick}>
        <span className="fr-connect__login">{loginLabel}</span>
        <span className="fr-connect__brand">{connectType}</span>
      </button>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
