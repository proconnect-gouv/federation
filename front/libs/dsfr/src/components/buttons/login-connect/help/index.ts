export * from './login-connect.help';
