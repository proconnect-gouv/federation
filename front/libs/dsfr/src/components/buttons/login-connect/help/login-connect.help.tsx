import React from 'react';

import { Strings } from '@fc/common';
import { t } from '@fc/i18n';

import { ConnectTypes } from '../../../../enums';

interface LoginConnectHelpProps {
  showIcon?: boolean;
  connectType: ConnectTypes;
}

export const LoginConnectHelp = React.memo(
  ({ connectType, showIcon = false }: LoginConnectHelpProps) => {
    const target = showIcon ? '_blank' : undefined;
    const rel = showIcon ? 'noreferrer' : undefined;

    const url =
      connectType === ConnectTypes.FRANCE_CONNECT
        ? 'https://franceconnect.gouv.fr/'
        : 'https://proconnect.gouv.fr/';

    const titleSuffix = showIcon
      ? `${Strings.WHITE_SPACE}${Strings.DASH}${Strings.WHITE_SPACE}${t('FC.Common.newWindow')}`
      : '';

    const helpLabel = t('DSFR.button.loginConnect.whatIs', { connectType });

    return (
      <p>
        <a href={url} rel={rel} target={target} title={`${helpLabel}${titleSuffix}`}>
          {helpLabel}
        </a>
      </p>
    );
  },
);

LoginConnectHelp.displayName = 'LoginConnectHelp';
