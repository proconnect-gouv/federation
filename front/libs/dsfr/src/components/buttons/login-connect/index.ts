export * from './login-connect.component';
