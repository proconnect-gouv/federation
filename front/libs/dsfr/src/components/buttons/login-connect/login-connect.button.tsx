import classnames from 'classnames';
import React from 'react';

import { ButtonTypes, ConnectTypes } from '../../../enums';

interface LoginConnectButtonProps
  extends Pick<React.HTMLProps<HTMLButtonElement>, 'onClick' | 'className'> {
  showIcon?: boolean;
  showHelp?: boolean;
  type?: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({
    className,
    connectType,
    onClick,
    showHelp = false,
    showIcon = false,
    type = ButtonTypes.BUTTON,
  }: LoginConnectButtonProps) => {
    const target = showIcon ? '_blank' : undefined;
    const rel = showIcon ? 'noreferrer' : undefined;
    const url =
      connectType === ConnectTypes.FRANCE_CONNECT
        ? 'https://franceconnect.gouv.fr/'
        : 'https://agentconnect.gouv.fr/';
    const label = `Qu’est ce que ${connectType} ?`;
    return (
      <div className={classnames('fr-connect-group', className)} data-testid="LoginConnectButton">
        <button className="fr-connect" type={type} onClick={onClick}>
          <span className="fr-connect__login">S’identifier avec</span>
          <span className="fr-connect__brand">{connectType}</span>
        </button>
        {showHelp && (
          <p>
            <a href={url} rel={rel} target={target} title={`${label} - nouvelle fenêtre`}>
              {label}
            </a>
          </p>
        )}
      </div>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
