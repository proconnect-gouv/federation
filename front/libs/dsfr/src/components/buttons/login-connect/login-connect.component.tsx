import classnames from 'classnames';
import React from 'react';

import type { PropsWithClassName, PropsWithOnClick } from '@fc/common';

import type { ConnectTypes } from '../../../enums';
import { ButtonTypes } from '../../../enums';
import { LoginConnectButton } from './button';
import { LoginConnectHelp } from './help';

interface LoginConnectComponentProps extends PropsWithClassName, PropsWithOnClick {
  showIcon?: boolean;
  showHelp?: boolean;
  type?: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectComponent = React.memo(
  ({
    className,
    connectType,
    onClick,
    showHelp = false,
    showIcon = false,
    type = ButtonTypes.BUTTON,
  }: LoginConnectComponentProps) => (
    <div className={classnames('fr-connect-group', className)} data-testid="LoginConnectComponent">
      <LoginConnectButton connectType={connectType} type={type} onClick={onClick} />
      {showHelp && <LoginConnectHelp connectType={connectType} showIcon={showIcon} />}
    </div>
  ),
);

LoginConnectComponent.displayName = 'LoginConnectComponent';
