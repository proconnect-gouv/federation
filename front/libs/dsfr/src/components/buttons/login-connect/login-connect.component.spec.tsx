import { render } from '@testing-library/react';

import type { ButtonTypes, ConnectTypes } from '../../../enums';
import { LoginConnectButton } from './button';
import { LoginConnectHelp } from './help';
import { LoginConnectComponent } from './login-connect.component';

jest.mock('./help/login-connect.help');
jest.mock('./button/login-connect.button');

describe('LoginConnectComponent', () => {
  // Given
  const onClickMock = jest.fn();
  const connectTypeMock = Symbol('connectType') as unknown as ConnectTypes;

  it('should match snapshot, with default props', () => {
    // When
    const { container } = render(
      <LoginConnectComponent connectType={connectTypeMock} onClick={onClickMock} />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('data-testid', 'LoginConnectComponent');
    expect(LoginConnectButton).toHaveBeenCalledOnce();
    expect(LoginConnectButton).toHaveBeenCalledWith(
      {
        connectType: connectTypeMock,
        onClick: onClickMock,
        type: 'button',
      },
      {},
    );
    expect(LoginConnectHelp).not.toHaveBeenCalled();
  });

  it('should match snapshot, with optionnal props', () => {
    // Given
    const buttonTypeMock = Symbol('button-type-mock') as unknown as ButtonTypes;

    // When
    const { container } = render(
      <LoginConnectComponent
        showHelp
        showIcon
        className="any-classname-mock"
        connectType={connectTypeMock}
        type={buttonTypeMock}
        onClick={onClickMock}
      />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('data-testid', 'LoginConnectComponent');
    expect(container.firstChild).toHaveClass('fr-connect-group any-classname-mock');
    expect(LoginConnectButton).toHaveBeenCalledOnce();
    expect(LoginConnectButton).toHaveBeenCalledWith(
      {
        connectType: connectTypeMock,
        onClick: onClickMock,
        type: buttonTypeMock,
      },
      {},
    );
    expect(LoginConnectHelp).toHaveBeenCalled();
    expect(LoginConnectHelp).toHaveBeenCalledWith(
      {
        connectType: connectTypeMock,
        showIcon: true,
      },
      {},
    );
  });
});
