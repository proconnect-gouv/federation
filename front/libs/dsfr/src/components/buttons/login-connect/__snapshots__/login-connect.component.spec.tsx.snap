// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginConnectComponent should match snapshot, with default props 1`] = `
<div>
  <div
    class="fr-connect-group"
    data-testid="LoginConnectComponent"
  >
    <div
      data-mockid="LoginConnectButton"
    />
  </div>
</div>
`;

exports[`LoginConnectComponent should match snapshot, with optionnal props 1`] = `
<div>
  <div
    class="fr-connect-group any-classname-mock"
    data-testid="LoginConnectComponent"
  >
    <div
      data-mockid="LoginConnectButton"
    />
    <div
      data-mockid="LoginConnectHelp"
    />
  </div>
</div>
`;
